Terminal.Gui/Views/Scroll.cs

```diff
@@ -0,0 +1,387 @@
+﻿namespace Terminal.Gui;
+
+/// <summary>
+///     Represents the "inside part" of a scroll bar, minus the arrows.
+/// </summary>
+public class Scroll : View
+{
+    /// <inheritdoc/>
+    public Scroll ()
+    {
+        WantContinuousButtonPressed = true;
+        CanFocus = false;
+        Orientation = Orientation.Vertical;
+        Width = Dim.Auto (DimAutoStyle.Content, 1);
+        Height = Dim.Auto (DimAutoStyle.Content, 1);
+
+        _slider = new ()
+        {
+            Id = "slider",
+            Width = Dim.Auto (DimAutoStyle.Content),
+            Height = Dim.Auto (DimAutoStyle.Content)
+        };
+        Add (_slider);
+
+        Added += Scroll_Added;
+        Removed += Scroll_Removed;
+        Initialized += Scroll_Initialized;
+        DrawContent += Scroll_DrawContent;
+        MouseEvent += Scroll_MouseEvent;
+        _slider.DrawContent += Scroll_DrawContent;
+        _slider.MouseEvent += Slider_MouseEvent;
+    }
+
+    private readonly View _slider;
+
+    private int _lastLocation = -1;
```

Comment:
Practically the behavior will be identical. If it is nullable I will have to check if it is null or not, because it works like a switch. The value is reset when the mouse is released. But if you insist I change it to nullable.